src/Orchard/Tasks/Locking/Services/DistributedLockService.cs

```diff
@@ -121,7 +120,9 @@ public class DistributedLockService : Component, IDistributedLockService {
 
                         if (throwOnTimeout)
                             throw new TimeoutException(String.Format("Failed to acquire lock '{0}' within the specified timeout ({1}).", internalName, timeout));
-
+                        else
+                            Monitor.Exit(monitorObj);
```

Comment:
Shouldn't it be released even if `throwOnTimeout` is true?

Reply:
@sebastienros the enclosing catch releases it further below see here: https://github.com/OrchardCMS/Orchard/blob/af0f7157eddd469427b0fe67848b07f024a2152c/src/Orchard/Tasks/Locking/Services/DistributedLockService.cs#L133

Reply:
@sebastienros yes, when an exception is thrown, explicitly or not, this is the enclosing catch that does a `Monitor.Exit()`, so we should not do it before if `throwOnTimeout` is true as we throw explicitly an exception, but we should do it if `throwOnTimeout` is false, i think we missed this path when the `throwOnTimeout` option has been added